arch/arm/boot/dts/overlays/revpi-flat-s-2022-overlay.dts

```diff
@@ -0,0 +1,378 @@
+				};
+				a3_red {
+					gpios = <&expander 5 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a3_red";
+				};
+				a4_green {
+					gpios = <&expander 6 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a4_green";
+				};
+				a4_red {
+					gpios = <&expander 7 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a4_red";
+				};
+				a5_green {
+					gpios = <&expander 8 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a5_green";
+				};
+				a5_red {
+					gpios = <&expander 9 GPIO_ACTIVE_LOW>;
+					linux,default-trigger = "a5_red";
+				};
+				led-act {
+					status = "disabled";
+				};
+			};
+
+			gpio-keys {
+				compatible = "gpio-keys";
+				pinctrl-names = "default";
+				pinctrl-0 = <&button_pins>;
+				user-event {
+					label = "User event key";
+					gpios = <&gpio 23 GPIO_ACTIVE_LOW>;
+					linux,code = <KEY_UNKNOWN>;
+				};
+			};
+
+			sdxxx-wlan {
+				cfg80211_wext = <0xf>;
+				cal_data_cfg = "none";
+				max_vir_bss=<1>;
+				ps_mode=<1>;
+				auto_ds=<1>;
+				host_mlme=<1>;
+				fw_name="nxp/sdiouartiw416_combo_v0.bin";
+			};
+		};
+	};
+
+	fragment@1 {
+		target = <&gpio>;
+		__overlay__ {
+			pinctrl-names = "default";
+			pinctrl-0 = <&relais_pins>;
+
+			spi5_cs_pins {
+				/* tpm ksz*/
+				brcm,pins     = <17 18>;
+				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			spi6_cs_pins {
+				/* aout ain mbus */
+				brcm,pins     = <6 10 27>;
+				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			tpm_pins: tpm_pins {
+				/* irq */
+				brcm,pins     = <22>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN>;
+				// TODO: ask HW about external pull up
+				brcm,pull     = <BCM2835_PUD_UP>;
+			};
+			mbus_pins: mbus_pins {
+				/* irq */
+				brcm,pins     = <45>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			button_pins: button_pins {
+				brcm,pins     = <23>;
+				brcm,function = <BCM2835_FSEL_GPIO_IN>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			relais_pins: relais_pins {
+				brcm,pins     = <28>;
+				brcm,function = <BCM2835_FSEL_GPIO_OUT>;
+				brcm,pull     = <BCM2835_PUD_OFF>;
+			};
+			rs485_0_pins: rs485_0_pins {
+				/* tx rx rts */
+				brcm,pins     = <4 5 7>;
+				brcm,function = <BCM2835_FSEL_ALT4>;
+				// TODO: ask HW about external pulls
+				brcm,pull     = <BCM2835_PUD_DOWN
+						 BCM2835_PUD_OFF
+						 BCM2835_PUD_UP>;
+			};
+			rs485_1_pins: rs485_1_pins {
+				/* tx rx rts */
+				brcm,pins     = <8 9 11>;
+				brcm,function = <BCM2835_FSEL_ALT4>;
+				// TODO: ask HW about external pulls
+				brcm,pull     = <BCM2835_PUD_DOWN
+						 BCM2835_PUD_OFF
+						 BCM2835_PUD_UP>;
+			};
+			i2c1_pins: i2c1_pins {
+				brcm,pins = <2 3>;
+				brcm,function = <BCM2835_FSEL_ALT0>;
+			};
+			sdhci_pins: sdhci_pins {
+				brcm,pins = <34 35 36 37 38 39>;
+				brcm,function = <BCM2835_FSEL_ALT3>;
+				brcm,pull = <BCM2835_PUD_OFF
```

Comment:
We don't need any pull-ups here because we have external pull-ups attached to DATA[0-3] and CMD. It seems that ther should be no pull-up at the CLK line. So we don't need any pull here.